feat(usb-uart): add FT232RL USB‑UART bridge and wire RX/TX to MCU

Iam-Madman opened the pull request:
Added the FT232RL USB‑to‑UART interface IC (FT232RL) into the schematic and powered it from the 3.3 V and 5 V rails. Exposed its TTL-level RXD and TXD pins through the hierarchical sheet and connected it to the MCU

- Placed FT232RL SSOP‑28 symbol and footprint, tied VCCIO to 3.3 V, VCC to USB_VCC, and GND to ground.
- Left legacy RS‑232 control lines (DSR#, DCD#, RI#) and CBUS pins unconnected.
- Updated hierarchy:added sheet pins on USB_UART page.

This enables USB‑serial communication between host PC and STM32F030F4Px

Iam-Madman committed:
feat(usb-uart): add FT232RL USB‑UART bridge and wire RX/TX to MCU

Added the FT232RL USB‑to‑UART interface IC (FT232RL) into the schematic and powered it from the 3.3 V and 5 V rails. Exposed its TTL-level RXD and TXD pins through the hierarchical sheet and connected it to the MCU

- Placed FT232RL SSOP‑28 symbol and footprint, tied VCCIO to 3.3 V, VCC to USB_VCC, and GND to ground.
- Left legacy RS‑232 control lines (DSR#, DCD#, RI#) and CBUS pins unconnected.
- Updated hierarchy:added sheet pins on USB_UART page.

This enables USB‑serial communication between host PC and STM32F030F4Px

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/4b49d687-655b-4543-96b5-c75c40749b92)
Move this usb connector to USB UART sheet.

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/bbbb8a94-a464-4159-907f-743581cce110)
Using bus would be better than creating rats nest using wire

benjaminjamesxyz commented:
USB data lanes should be differential pairs

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/bbe5d863-6239-42e9-830d-05914ba633cb)
Make use of the reset pin so that when mcu boots up the chip starts from a known state.

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/060d222b-e8db-41ca-ad42-2ddc6c798eab)
Better to keep AGND in separate plane, power symbol GNDA in kicad can be used here.

Iam-Madman committed:
feat(usb-uart): add FT232RL for USB-serial, update schematic and layout

Added FT232RL USB‑to‑UART interface IC, powered from 3.3 V and 5 V rails. Connected RXD and TXD to MCU via hierarchical sheet. Left RS‑232 control lines and CBUS pins unconnected. Updated hierarchy with USB_UART sheet pins.

BREAKING CHANGE: USB connector moved to USB_UART sheet; schematic wiring changed to bus-based layout. AGND now kept in separate plane using GNDA power symbol in KiCad.

Iam-Madman commented:
changes have been addressed **except the reset pin** configuration so that when mcu boots up the chip starts from a known state and the differential pairing of **USB data lanes**

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/cb3ad5f4-2453-4bf0-9afc-015c341c81bd)
Check the directions of the hierarchical labels, it should match the signal direction.

benjaminjamesxyz commented:
> changes have been addressed **except the reset pin** configuration so that when mcu boots up the chip starts from a known state and the differential pairing of **USB data lanes**

may be reset can be left for next revision of design

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/ce6158db-2536-45fa-b304-ebca599ed863)
Separating the resistors and using labels will be a cleaner approach.
note that, what you have done is actually correct.

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/187f77af-901f-4ede-80af-967277cbc9de)
There is a 0603 SMD package available for the USB CC pin resistors.
Make sure you have attached the proper footprint to the symbols. Use the hand solder ones.

Iam-Madman committed:
fix: Separted CC1 and CC2 resistors and fixed the UART label directions

Iam-Madman committed:
feat: added footprints for the 5.1k resistors

Iam-Madman committed:
feat(FT232RL_Reset): Added FT232RL Reset functionality using the Reset switch

Iam-Madman committed:
fix: Fixed the power label shape in power tree

Iam-Madman committed:
feat(LEDs): Added LEDs to the USB UART

Added a power LED and RX TX LEDs to the CBUS pins

Iam-Madman committed:
feat(LED_Footprints): Added Footprints to the LEDs and its Resistors

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/7bfd4b44-aee3-4d9b-ab49-3a1e9e03b45a)
@Iam-Madman still the Hierarchical Label for power is not changes
All Hierarchical Label for power should be passive

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/4a49db6b-5289-40b8-8642-ea31a8ddfb10)
All though what you have done is correct. You could are some minor context,
so that its not necessary to go back to datasheet refer everything. For details one can refer the datasheet.

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/17229765-3caa-4415-90be-70449f959ecf)
Would be better if the label conveyed, for what the UART is actually used for.

Iam-Madman committed:
fix(Power_tree): Hierarchical Label is made passive

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/d33a2954-4c60-41d1-8424-0f6f75c56573)
If its for board power I should be in power tree sheet. If its for USB to UART chip status, connect it to appropriate status pin.

Iam-Madman committed:
fix(MCU): Hierarchical Label is made passive

Iam-Madman committed:
fix(USB_UART): Hierarchical Label is made passive

Iam-Madman committed:
fix(USB_UART): Removed an unnecessary datasheet reference

Iam-Madman committed:
fix(Power_led): Moved the power led to power tree sheet from USB UART sheet

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/8dfa4138-0aaa-46a2-a790-ae3bc426795b)
power status led should be connected at the beginning of power tree. To properly show power on status.

Iam-Madman committed:
fix(USB_UART): Fixed the USB_UART bus labels in the root sheet

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/9a7a85ce-f36b-4810-9470-7a525c724da2)
Instead of just remove the reference to datasheet add some text why its connected like this.

Iam-Madman committed:
fix(USB_UART): Minor fix in the label

Iam-Madman committed:
feat(CC1_CC2_Resistors): Added a text to explain why the resistors of the following value is used

benjaminjamesxyz commented:
![image](https://github.com/user-attachments/assets/910d2810-c1f2-4ddd-b0b8-34ca8f6340e1)
Give label names in terms of MCU, for all labels.
Hierarchical Labels should be renamed too.

Iam-Madman committed:
fix(USB_UART): minor label fixes

Iam-Madman committed:
feat(POWER_LED): Changed the position of the power led to the beginning of the power tree

benjaminjamesxyz commented:
looks okay. Further improvements can be done later if required.
merging it for now.

benjaminjamesxyz reviewed (approved):
looks good
